### PR TITLE
Reference app crash when country is null.

### LINKED ISSUE #774: Reference app crashed after opening it
**Describe the bug**
The reference app throws NPE error on the country value after opening it

**Component**
reference app

**To Reproduce**
1. Open the 'reference app'
2. Crash

**Expected behavior**
It should not crashed

**Screenshots**
-

**Smartphone (please complete the following information):**
 - Device: [Pixel4 emulator]
 - Android version: [11]
 - Build number: [RSR1.201013.001]

**Additional context**
Error log
```
2021-09-05 19:24:39.484 17706-17706/com.google.android.fhir.reference E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.google.android.fhir.reference, PID: 17706
    java.lang.NullPointerException: country must not be null
        at com.google.android.fhir.reference.PatientListViewModelKt.toPatientItem(PatientListViewModel.kt:145)
        at com.google.android.fhir.reference.PatientListViewModel.getSearchResults(PatientListViewModel.kt:71)
        at com.google.android.fhir.reference.PatientListViewModel$getSearchResults$1.invokeSuspend(Unknown Source:12)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Would you like to work on the issue?**
N.A

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #774 

**Description**
Country element value was null.

**Alternative(s) considered**
no

**Type**
Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt [hunk 1]
```diff
@@ -129,8 +129,8 @@ internal fun Patient.toPatientItem(position: Int): PatientListViewModel.PatientI
   val gender = if (hasGenderElement()) genderElement.valueAsString else ""
   val dob = if (hasBirthDateElement()) birthDateElement.valueAsString else ""
   val phone = if (hasTelecom()) telecom[0].value else ""
-  val city = if (hasAddress()) address[0].city else ""
-  val country = if (hasAddress()) address[0].country else ""
+  val city = if (hasAddress() && address[0].hasCity()) address[0].city else ""
+  val country = if (hasAddress() && address[0].hasCountry()) address[0].country else ""
```
Comment (aditya-07) [inline reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt:133]: Other fields can be null, so should check them as well. 
We could do this while constructing PatientListViewModel.PatientItem
```
   return PatientListViewModel.PatientItem(
    id = position.toString(),
    resourceId = patientId,
    name = name,
    gender = gender ?: "",
    dob = dob ?: "",
    phone = phone ?: "",
    city = city ?: "",
    country = country ?: "",
    isActive = isActive,
    html = html
  )
```

Don't forget to modify `PatientItemViewHolder` to handle null/empty values.

**Review by aditya-07 — CHANGES_REQUESTED** [review]

File: reference/src/main/java/com/google/android/fhir/reference/PatientItemViewHolder.kt [hunk 2]
```diff
@@ -41,17 +41,25 @@ class PatientItemViewHolder(binding: PatientListItemViewBinding) :
...
       }
     }
+  }
 
   /** The new ui just shows shortened id with just last 3 characters. */
   private fun getTruncatedId(patientItem: PatientListViewModel.PatientItem): String {
+    if (patientItem.resourceId.length < 3) {
+      return patientItem.resourceId
+    }
     return patientItem.resourceId.substring(patientItem.resourceId.length - 3)
```
Comment (jingtang10) [inline reference/src/main/java/com/google/android/fhir/reference/PatientItemViewHolder.kt:63]:
```suggestion
    return patientItem.resourceId.takeLast(3)
```

File: reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt [hunk 3]
```diff
@@ -138,11 +137,11 @@ internal fun Patient.toPatientItem(position: Int): PatientListViewModel.PatientI
     id = position.toString(),
     resourceId = patientId,
     name = name,
-    gender = gender,
-    dob = dob,
-    phone = phone,
-    city = city,
-    country = country,
+    gender = gender ?: "",
+    dob = dob ?: "",
```
Comment (jingtang10) [inline reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt:141]: i'm not sure this should be a string... date format varies hugely in different locales and system settings probably should be respected...

please create a separate ticket.

**Review by jingtang10 — APPROVED** [review]

File: reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt (hunk 3 above)
Comment (santosh-pingle) [reply reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt:141]: Actually, this data class is for item view only to show the patient attributes on the screen, thats why we are seeing all fields as string also  same was updated in doc in existing code.

**Review by aditya-07 — APPROVED** [review]
LGTM. Thanks Santosh!

**Pull request merged**